### DISCUSSION TITLE
[RocketMQ 5.1.4] 内外网隔离的情况下，怎么做到内外网都可以通过MQ收发消息

### DISCUSSION THREAD

**Discussion by maopuhua** [apache/rocketmq#7558] [General]
RocketMQ单机部署，namesvr和broker都部署在DMZ区域机器，此机器有内外网两个IP地址，其他内网机器只能识别内网IP，外网机器只能访问外网IP。 如果将broker.conf中brokerIP1配置为外网IP地址，内网无法获取到topic的路由信息。 如果是将brokerIP1配置为内网IP地址，外网机器又会同样提示无法获取topic的路由信息。

请问下broker是否有办法支持配置多IP的情况。

**Comment by jimchanghub** [comment]
我也碰到同样的问题了，请问有办法解决吗

**Comment by irisrain** [comment]
同样问题。5x好像通过proxy的一个功能解决了，我是4.6。。不知道怎么解决

**Reply by gzlboy** [reply]
对，最新版proxy对云原生环境就很友好了，目前我还没测通，proxy通过nodeport暴露，不知道为何集群外连接不上

**Reply by gdhua** [reply]
这几天搞5x的，试proxy，还是不行

**Reply by pxd98** [reply]
proxy应该也是一样，把自己的brokerIP写到topic route里发给客户端，可以试一下proxy里的IP配外网IP，外网客户端连proxy。内网客户端连nameserver，不连proxy

**Comment by maopuhua** [comment]
暂时没有找到解决方案，后面是切换到了rabbitmq来解决的

**Comment by pxd98** [comment]
能否把brokerIP设置为域名，然后加hosts或者dns解析来解决？
